llvm/include/llvm/MC/MCDecoder.h

```diff
@@ -62,6 +72,13 @@ insertBits(IntType &field, IntType bits, unsigned startBit, unsigned numBits) {
   field |= bits << startBit;
 }
 
+// InsnBitWidth is essentially a type trait used by the decoder emitter to query
+// the supported bitwidth for a given type. But default, the value is 0, making
+// it an invalid type for use as `InsnType` when instantiating the decoder.
+// Individual targets are expected to provide specializations for these based
+// on their usage.
+template <typename T> inline constexpr uint32_t InsnBitWidth = 0;
```

Comment:
This one *could* be `static` as well (unlike specializations, it should have the same value in all translation units, so `inline` works here).
Perhaps it *should* be made `static` anyway to prevent accidental copy&paste bugs when specializing it.